src/windows/mapProxy.js

```diff
@@ -167,11 +171,153 @@
         Microsoft.Maps.loadModule('Microsoft.Maps.Map', {callback: initMap, culture: "en-us", homeRegion: "AU"});
     };
 
+    // http://stackoverflow.com/a/9458996
+    function arrayBufferToBase64(buffer) {
+        var binary = "";
+        var bytes = new Uint8Array(buffer);
+        var len = bytes.byteLength;
+        var i;
+
+        for (i = 0; i < len; i += 1) {
+            binary += String.fromCharCode(bytes[i]);
+        }
+        return window.btoa(binary);
+    }
+
+    function buildStaticMapUrl(opts) {
+        var url = "https://dev.virtualearth.net/REST/v1/Imagery/Map/";
+        var i;
+
+        url += opts.imagerySet;
+        url += (opts.centerPoint) ? "/" + opts.centerPoint.toString() : "";
+        url += (opts.zoomLevel) ? "/" + opts.zoomLevel : "";
+
+        url += "?key=" + opts.key;
+        url += "&format=" + opts.format;
+
+        if (opts.pushpins) {
+            for (i = 0; i < opts.pushpins.length; i += 1) {
+                url += "&pushpin=" + opts.pushpins[i].slice(0,2).toString();
+                url += (opts.pushpins[i][2]) ? ";" + opts.pushpins[i][2]: "";
+                url += (opts.pushpins[i][3]) ? ";" + opts.pushpins[i][3]: "";
+            }
+        }
+
+        url += (opts.declutterPins) ? "&declutterPins=1" : "";
+        url += (opts.mapSize) ? "&mapSize=" + opts.mapSize.toString() : "";
+
+        return url;
+    }
+
+    map.getStaticMap = function (onSuccess, onError, options) {
+        var opts = {};
```

Comment:
Just curious, do we need to do all the validation of options and then rebuilding the object? 

I am assuming bing does all these checks itself and returns an error, so why not just pass through the options value, that way if bing changes the options (eg adds a new option/removes an old one) we don't have to change this ?